Use a secret for db credentials (#9)
plone/helm-charts: plone6-classic-nodb/README.md

## Prerequisites

An external PostgreSQL database cluster is needed for this setup to work.
The external database can be located inside or outside of your Kubernetes cluster, or the namespace you use to install this chart, as long as it is accessible.

> You will need to access to the PostgreSQL database cluster for Plone, see the next section.

Reviewer: ```suggestion
You will need access to the PostgreSQL database cluster for Plone.
See the next section.
```

Author: argh there's a missing 'configure' in my sentence, it should be: 'You will need to configure access to the ...'. Does that have an impact on your suggestion?

Reviewer: Updated suggestion.

An external database is needed for this setup to work.
The external database can be located inside or outside of your Kubernetes cluster, or the namespace you use to install this chart, as long as it is accessible.

> Database credentials (database name, user name, password) are expected in a secret.

Reviewer: Is host required as well, or is localhost assumed as default?

```suggestion
Database credentials (database name, user name, and password) are expected in a secret.
```

Author: PostgreSQL database cluster host and port are not configured with the secret. I have tried to elaborate on the database configuration in the README. Imo database cluster host and port should not be secret, they are not part of the credentials needed to access the actual database (you can move the database between database clusters without having to change the credentials). Maybe we need to this discuss this later, but let's start like this.

Reviewer: I don't have an opinion. I wanted to make sure that the omission was deliberate, not an oversight, and that the host was defined somewhere. I don't know how to connect to a database without a host parameter.

Should the host configuration be applied to the other three README.mds?

Author: > Should the host configuration be applied to the other three README.mds?

No, at least not in the same fashion. The other charts come with an included database, so there is no need to configure host + port.

The secret name defaults to `plonedb`.

> [Create the secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret) e.g. using `kubectl`:

Reviewer: ```suggestion
Using the `kubectl` command, you can create a secret.
```

```shell
kubectl create secret generic plonedb \
    --from-literal=database-name='plone' \
    --from-literal=database-user='plone' \
    --from-literal=database-password='plone'
```

You can override the default secret name in `values.yaml`:

```yaml
backend:
  dbsecret:
    name: <your-secret>
```

> 

Reviewer: ```suggestion

See also the Kubernetes documentation [Create a Secret](https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl/#create-a-secret).

```

### Backend

You will need to configure the following options in `values.yaml` or on the command line:

```yaml
backend:
  databasehost: your.db.host
```
The PostgreSQL database cluster host.

> There is no sensible default for this option, you will have to modify it.

Reviewer: ```suggestion
There is no sensible default for this option.
You will have to modify it.
```
